### DISCUSSION TITLE
Solid background?

### DISCUSSION THREAD

**Discussion by strider72** [chillerlan/php-qrcode#197] [Q&A] [answered]
This seems it should be silly simple, but... How do I make a QR with a solid white background?

I'm using circles for the modules, but with the exception of Jpeg format, any QR I make has dots for the light areas, or nothing at all. How do I generate a QR image with a solid background?

(I'm using v5 beta)

**Comment by codemasher** [comment]
Hey, I've just recently cleaned up the background color situation for GD and Imagick output. You should be able to set `QROptions::$bgColor` to a value for the respective output and `QROptions::$drawLightModules` to `false`. The background color setting doesn't work for SVG as you can set this via CSS.

**Reply by codemasher** [reply, marked as answer]
Example for GD:

```php
$options = new QROptions([
	'outputType'          => QROutputInterface::GDIMAGE_PNG,
	'bgColor'             => [200, 150, 200],
	'imageTransparent'    => false,
	'drawLightModules'    => false,
]);
```

**Comment by strider72** [comment]
Got it. I was a) mainly testing with SVG, and b) missing `imageTransparent`.

May I make a request? Make SVG work the same as the others? If I specifically say to add a background, it should have a background.

I — in part — use this as a utility so people I work with can generate branded QR codes. I may know how to use CSS, but I don't expect my layman users to.

(And now I'm off to figure out how to add a logo!)

Very cool library, if sometimes frustrating.... :-)

**Comment by strider72** [comment]
I take it back. Saved an SVG to file and can't find the "box" I need to style with CSS.

Even if set to transparent, the background "box" should be there...?

**Comment by strider72** [comment]
I'll make my request a bit more precise.  With SVG format, please:

1) add a containing square that is behind all other elements, sized to the outer edge of the Quiet Zone, and
2) if image is not transparent, set the box to the background color. If image is transparent, box is either transparent or just `display:none`

I'm setting my QRs to circular modules, and can't find any way to have a solid white background behind the visible (dark) circles. A simple box to style would simplify things greatly.

**Reply by strider72** [reply]
I've made this request a separate discussion thread
